stellargraph/mapper/sampled_node_generators.py

```diff
@@ -563,3 +564,85 @@ def flow_from_dataframe(self, node_ids):
 
         """
         return NodeSequence(self.sample_features, self.batch_size, node_ids.index)
+
+
+class Node2VecNodeGenerator(BatchedNodeGenerator):
+    """
+    A data generator for node representation prediction with node2vec models.
+
+    At minimum, supply the StellarGraph and the batch size.
+
+    The supplied graph should be a StellarGraph object that is ready for
+    machine learning. Currently the model does not require node features for
+    nodes in the graph.
+
+    Use the :meth:`flow` method supplying the nodes to get an object
+    that can be used as a Keras data generator.
+
+    Example::
+
+        G_generator = Node2VecNodeGenerator(G, 50)
+        data_gen = G_generator.flow(node_ids)
+
+    Args:
+        G (StellarGraph): The machine-learning ready graph.
+        batch_size (int): Size of batch to return.
+        name (str or None): Name of the generator (optional).
+    """
+
+    def __init__(self, G, batch_size, name=None):
+        super().__init__(G, batch_size, use_node_features=False)
+        self.name = name
+
+    def sample_features(self, head_nodes, batch_num):
+        """
+        Get the ids of the head nodes, and return these as a list of feature
+        arrays for the node2vec algorithm.
+
+        Args:
+            head_nodes: An iterable of head nodes to perform sampling on.
+
+        Returns:
+            A list of feature arrays, with each element being the id of each
+            head node.
+        """
+
+        head_node_ids = self.graph._get_index_for_nodes(head_nodes)
+        batch_feats = np.array(head_node_ids)
+        return batch_feats
```

Comment:
similar to the same method in the link generator, this could be:
```suggestion
        return self.graph._get_index_for_nodes(head_nodes)
```
or if you did want to assign it to a named variable first, I think `iloc` would be a more accurate term than `ids` here.
```suggestion
        head_node_ilocs = self.graph._get_index_for_nodes(head_nodes)
        return head_node_ilocs
```

Reply:
Thanks Kevin! I have changed it to ``return self.graph._get_index_for_nodes(head_nodes)``